### PR TITLE
feat: Exception Autocapture

### PR DESCRIPTION
**Pull request by pauldambra**
## Changes

adds

* autocapture for exceptions
* autocapture for unhandled promise rejection
* manual exception capture 

- [x] ~~session recording URL~~
- [x] add an captureException method to PostHog instance
- [x] update sentry integration to send PostHog exception fields too

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -53,6 +53,7 @@ export interface PostHogConfig {
     ui_host: string | null
     token: string
     autocapture: boolean | AutocaptureConfig
+    exception_autocapture: boolean
```
Comment (pauldambra) [inline src/types.ts:56]:
```suggestion
    // this is currently in development and may have breaking changes without a major version bump
    exception_autocapture: boolean
```

File: src/extensions/exceptions/exception-autocapture.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            const errorProperties = errorToProperties(args)
+            const propertiesToSend = { ...properties, ...errorProperties }
+
+            const posthogHost = this.instance.config.ui_host || this.instance.config.api_host
+            if (this.instance.sessionRecordingStarted()) {
+                errorProperties.$exception_sessionRecordingURL =
+                    posthogHost +
+                    '/recordings/' +
+                    this.instance.sessionManager.checkAndGetSessionAndWindowId(true).sessionId
+            }
```
Comment (benjackwhite) [inline src/extensions/exceptions/exception-autocapture.ts:122]: We have `get_session_replay_url()` now which should be used here. 

One thought though - why do we want this 🤔 is it not redundant given that we can rebuild it on the other end?

File: src/posthog-core.ts [hunk 3]
```diff
@@ -104,6 +106,8 @@ const defaultConfig = (): PostHogConfig => ({
     ui_host: null,
     token: '',
     autocapture: true,
+    // TODO: change to undefined when we release this so that remote config can turn it on even if unconfigured by the user
+    exception_autocapture: false,
```
Comment (benjackwhite) [inline src/posthog-core.ts:110]: Feels a little weird that we have `exception_autocapture` locally and `autocaptureExceptions` remotely... Not sure if that means it should be different but it feels confusing...

**Review by benjackwhite — COMMENTED** [review]
Some first pass comments. Will take a proper look at the exception code in a bit

**Review by benjackwhite — COMMENTED** [review]
Some first pass comments. Will take a proper look at the exception code in a bit

**Review by benjackwhite — CHANGES_REQUESTED** [review]
Small changes but overall this looks great. Didn't review the "inspired" code too deeply but it all looks very clever ;)

**Pull request merged**